src/store/actions/authActions.js

```diff
@@ -0,0 +1,56 @@
+import { signupUser, loginUser } from '@/api/api';
+
+import {
+  setUserDetails,
+  setUserLoggedIn,
+  logOutUser,
+} from '../reducers/authSlice';
+
+export const signUpAction = (userData, setBtnLoader) => {
+  return async (dispatch) => {
+    try {
+      setBtnLoader(true);
+      const { data } = await signupUser(userData);
+      const userDetails = {
+        name: data.name,
+        email: data.email,
+      };
+      dispatch(setUserDetails(userDetails));
+      dispatch(setUserLoggedIn());
+
+      localStorage.setItem('token', data.token);
+    } catch (err) {
+      console.log(err);
+    }
+    setBtnLoader(false);
```

Comment:
make the button loader false in finally block

Reply:
update the try catch blocks